src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const cleanedString = sourceString.replace(/;\s*;/g, ';');
+  const regex = /\s*([^:\s]+)\s*:\s*([^;]+)\s*;?/g;
+  let match;
+
+  while ((match = regex.exec(cleanedString)) !== null) {
+    const key = match[1].includes(';', 0)
+      ? match[1].trim().slice(1)
+      : match[1].trim();
+    const value = match[2].includes(';', 0)
+      ? match[2].trim().slice(1)
+      : match[2].trim();
```

Comment:
The checks for ';' within the 'key' and 'value' are redundant due to the regex pattern used. The pattern already handles semicolons at the end of each declaration, so these checks can be removed.